pkg/scalers/kafka_scaler.go

```diff
@@ -317,7 +317,7 @@ func (s *kafkaScaler) getLagForPartition(partition int32, offsets *sarama.Offset
 
 	if consumerOffset == sarama.OffsetNewest || consumerOffset == sarama.OffsetOldest {
 		if s.metadata.offsetResetPolicy == latest {
-			lag = 0
+			lag = sarama.OffsetNewest
```

Comment:
I don't know the sarama library but there is something that sounds strange to me first of all in the first `if` statement:

`if consumerOffset == sarama.OffsetNewest || consumerOffset == sarama.OffsetOldest`

When a consumer is new in a consumer group there is no offset committed of course. Why the sarama fetch offset should return `OffsetNewest` or `OffsetOldest`? It could return the one or the other if it knew the reset policy asked by the consumer (so returning `OffsetNewest` if `latest` and `OffsetOldest` if `earliest`) but it doesn't know, right?
It's not actually clear to me the sarama behaviour here tbh.

Reply:
It is actually not Sarama deciding to return -1, but the kafka coordinator here: https://github.com/apache/kafka/blob/8bde3d476f23a9d9dd06b3360281a3a3fca2804e/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L481
From what I see, if no consumer group exists, or [no offset is committed](https://github.com/apache/kafka/blob/8bde3d476f23a9d9dd06b3360281a3a3fca2804e/core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala#L503): 
the coordinator will return the value of OffsetFetchResponse.INVALID_OFFSET that is -1:
https://github.com/apache/kafka/blob/6da70f9b95d2f7cc1f32de4e405661a8015bfa7e/clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchResponse.java#L53

Maybe we could introduce the concept of invalid offset instead of using offsetNewest and Oldest that are related to the consumer and not the admin client?

Reply:
Absolutely agree ... this -1 value has a different meaning them not related to what Sarama returns, so it's better having it separated.

Reply:
Do you think we should return an error if the offset is invalid? This would help identify the case of missing offsets, hence having a more significant condition also for the isActive method.

Reply:
absolutely yes, I don't like the `if` condition based on sarama newest and oldest + lag > 0 or lag != 0. An error should simplify and make it clearer.

Reply:
Done, I didn't know whether to log an error or an info message with a low verbosity level, since this is not a fatal error as  the other two in the getLagForPartition method.